### PR TITLE
BUGFIX: Cleanup media browser `addFlashMessageTrait`

### LINKED ISSUE #3898: BUG: FlashMessages - current escaping of tags / collections leads to an exception
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

It is possible to crash parts of the neos backend by just creating a collection or tag (management -> media) named `%`

ezgif-2-3616c7b01a

### Expected Behavior

Neos should not crash no matter what user input is made.

### Steps To Reproduce

- navigate to the media section
- create a tag or collection named `%`

### Environment

```markdown
- Neos: 8.0, 8.1
- PHP: 8.1
```

## notes
As Ferdinand Kuhl mentioned, there are some places where this happens. See e.g. the following line:
https://github.com/neos/media-browser/blob/master/Classes/Controller/TagController.php#L78

This behavior could also occur with other methods, I have only tested it in the media library.

### PR DESCRIPTION
**Pull request by mhsdesign**
resolves: https://github.com/neos/neos-development-collection/issues/3898

another fix for the problem, by refactoring the `AddFlashMessageTrait` 

thanks @crydotsnake for the initial effort with https://github.com/neos/neos-development-collection/pull/3940 but it if accepted it would make your approach obsolete. (I dindt wanted to force push on your branch - thus a new pr ^^)

see https://github.com/neos/neos-development-collection/pull/3940#issuecomment-1296346190 for the explanation of this fix

**Upgrade instructions**

**Review instructions**

Create a tag with `%` sign and wait for the flashmessage: 

image

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * @var \Neos\Flow\I18n\Translator
+     */
+    protected $_translator;
+
+    public function addTranslatedFlashMessage(
+        string $translationIdForTheMessageBody,
+        string $severity = Message::SEVERITY_OK,
+        array $messageArguments = [],
+        ?int $messageCode = null
+    ): void {
```
Comment (crydotsnake) [inline Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php]:
```php
public function addTranslatedFlashMessage(string $translationIdForTheMessageBody, string $severity = Message::SEVERITY_OK, array $messageArguments = [], ?int $messageCode = null): void
    {
// Your Code
```

I would combine everything in one line. That looks better.

File: Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        $messageBody = $this->_translator->translateById($translationIdForTheMessageBody, $messageArguments, null, null, 'Main', 'Neos.Media.Browser');
```
Comment (crydotsnake) [inline Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php]: `$translationIdForTheMessageBody` => `$translationMessageId` is shorter, and sounds better.

**Review by crydotsnake — CHANGES_REQUESTED** [review]
From the functionality point of view it looks good. I have only deposited a few notes regarding codestyle, and naming.

File: Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php]: yes while we could argue about this i think this is fine as is. (Allowed codestyle in PSR2 and PSR12 and our code style check doesnt complain)

File: Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php (hunk 2 above)
Comment (mhsdesign) [reply Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php]: I would argue that its a purely internal parameter name - so no real worries and we dont need to save characters.
So id like to stick with it and i actually like the descriptive name.

Previously it was `@param string $messageBody The translation id for the message body.`

Comment (mhsdesign) [reply Neos.Media.Browser/Classes/Controller/AddTranslatedFlashMessageTrait.php]: okay im changing it to `$messageBodyTranslationId`

**Comment by kdambekalns** [comment]
TBH: I don't see a difference in the code (except for naming). Is there any?

File: Neos.Media.Browser/Classes/Controller/AddFlashMessageTrait.php [hunk 3]
```diff
@@ -1,49 +0,0 @@
...
-     * @return void
-     */
-    public function addFlashMessage($messageBody, $messageTitle = '', $severity = Message::SEVERITY_OK, array $messageArguments = [], $messageCode = null): void
-    {
-        if (is_string($messageBody)) {
-            $messageBody = $this->_translator->translateById($messageBody, $messageArguments, null, null, 'Main', 'Neos.Media.Browser') ?: $messageBody;
-        }
-
-        $messageTitle = (string)$this->_translator->translateById($messageTitle, $messageArguments, null, null, 'Main', 'Neos.Media.Browser');
-        parent::addFlashMessage($messageBody, $messageTitle, $severity, $messageArguments, $messageCode);
```
Comment (mhsdesign) [inline Neos.Media.Browser/Classes/Controller/AddFlashMessageTrait.php:47]: @kdambekalns these lines where changed.

i removed `$messageTitle` as we dont use it and its confusing to translate something we dont use.

the real fix lies in `parent::addFlashMessage` where we dont pass in the (unneded - already interpolated) `$messageArguments` -> and thus prevent a `sprintf` call in `Flow\Message`

Comment (kdambekalns) [reply Neos.Media.Browser/Classes/Controller/AddFlashMessageTrait.php:47]: > the real fix lies in `parent::addFlashMessage` where we dont pass in the (unneded - already interpolated) `$messageArguments`

Ah! I was too focused on the `translate` call, it seems…

**Review by kdambekalns — APPROVED** [review]
by reading

**Review by crydotsnake — APPROVED** [review]

**Comment by crydotsnake** [comment]
Tested, and works!

**Pull request merged**